File: c/grains/Iteration4/grains.c

#include <math.h>

#include "grains.h"

uint64_t square(uint8_t index)
{
    if (index <= 0 || index > 64) return 0;
    
    return UINT64_C(1) << (index - 1);
}

uint64_t total(void)
{
    return ~square(64) | square(64);
}

File: csharp/grains/Iteration1/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64) throw new ArgumentOutOfRangeException();
        
        return (ulong)Math.Pow(2, n-1);
    }

    public static ulong Total() =>
        (ulong)Enumerable
            .Range(1, 64)
            .Select(n => (int)Square(n))
            .Sum();
}
